### PR TITLE
Delete from cart V2

### PR DESCRIPTION
**Pull request by bailiewu**
Delete from cart and carts now have quantities.

- Remove 1 item from cart.
- Reduce the quantity of an item in the cart by 1.
- Empty a user's cart.
- Correctly adds items to a user's cart (1 or many)
- included code for a sql view in readme that outerjoins orders with ordersproducts
- added a script to only build & seed.

`addToCart(userId: number, productId: number, quantity: number = 1)`
`decreaseProductQuantityFromCart(userId: number, productId: number, removeAll: boolean = false)`
`emptyCart(userId: number)`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "client:build": "./node_modules/.bin/tsc -p ./src/client/tsconfig.json && ./node_modules/.bin/webpack",
     "seed": "node js/server/seed.js",
     "seed-faker": "node js/server/seed-faker.js",
-    "server:watch": "./node_modules/.bin/tsc -w & ./node_modules/.bin/nodemon --ignore client --ignore src",
+    "seed-faker:dev": "npm run server:build && npm run seed-faker",
+    "server:watch": "./node_modules/.bin/tsc -w & ./node_modules/.bin/nodemon --ignore client --ignore js",
```
Comment (hgranthorner) [inline package.json]: this should be --ignore client -- ignore src

File: src/server/models/order.ts [hunk 2]
```diff
@@ -32,30 +32,107 @@ class Order extends Model<Order> {
...
+            let foundProductFromCart = userOrderCart.find(orderProdLineItem => orderProdLineItem.productId === productId)
+            if (!foundProductFromCart) {
+              // product does not exist in cart. add product to order
+              return await OrdersProducts.create({ orderId: cart.id, productId: productId, quantity: quantity })
+            } else {
+              // product exists in cart. increase quantity
+              return await foundProductFromCart.update({ quantity: foundProductFromCart.quantity + quantity })
+            }
+          })
+          .then(() => cart.id)
```
Comment (hgranthorner) [inline src/server/models/order.ts]: Have you tested this? Can it really find cart? I thought it would be out of scope by now

**Review by hgranthorner — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (bailiewu) [reply package.json]: Good catch, i may have not seen this due to a merge conflict. I didnt even think i touched this. Probably from an old commit

File: src/server/models/order.ts (hunk 2 above)
Comment (bailiewu) [reply src/server/models/order.ts]: Yes, this is tested. Console logged where cart.id was initialized and then at the end where it is returned results in the same value.

**Review by hgranthorner — APPROVED** [review]

**Pull request merged**